src/coronawhy_vt/utils/local_processing.py

```diff
@@ -0,0 +1,375 @@
+    return {'sha': paper_id,
+            'blockid':
+            block_id,
+            'word': found_words,
+            'sec': section,
+            'pos': positions,
+            'block': block_text}
+
+# Recommend changing expected block argument with the entire of the processed
+# paper as it seems the only functionality for this is updating the text of
+# the section of paper to the underscore tokens
+
+
+def run_replace(block: str, replace_dict: dict) -> str:
+    """Replace routine for dealing with n-grams.
+
+    :param block: Contains the text block replacement is done on
+    :param replace_dict: {token: replacement_token}
+
+    :return: Occurrences of token replaced with replacement_token
+    """
+    for k in replace_dict.keys():
+        if k in block:
+            block = block.replace(k, replace_dict[k])
+    return block
+
+
+# TODO These paths are kaggle paths, replace first with github paths then DataVerse paths once setup
+paths = ["/kaggle/input/CORD-19-research-challenge/noncomm_use_subset/noncomm_use_subset/pdf_json/",
+         "/kaggle/input/CORD-19-research-challenge/comm_use_subset/comm_use_subset/pdf_json/",
+         "/kaggle/input/CORD-19-research-challenge/biorxiv_medrxiv/biorxiv_medrxiv/pdf_json/",
+         "/kaggle/input/CORD-19-research-challenge/custom_license/custom_license/pdf_json/"]
+# TODO Right now this function references an externally declared variable, suggest either moving
+# into a class or make paths an argument
+# Also function name might need to change, a little ambiguous
+
+
+def extract_to_csv(
+        words: list,
+        file_name: str,
+        lemmatized: bool = True,
+        run_title: bool = True,
+        run_abstract: bool = True,
+        run_body: bool = False,
+        replace_dict: dict = None) -> None:
+    """Store aggregate features of vocab words in the main CORD-19 dataset.
+
+    :param words: List of strings representing words of interest
+    :param file_name: Path of the location to store final csv
+    :param lemmatized: If true, perform lemmatized feature extraction
+    :param run_title: If true, features extracted from the title will be included in the aggregate
+    :param run_abstract: If true, features extracted from the abstract will be included in the aggregate
+    :param run_body: If true, features extracted from the body will be included in the aggregate
+    :param replace_dict (dict): {token: replacement_token}
+    """
+    if lemmatized:
+        pull_mentions = pull_mentions_lemmatized
+    else:
+        pull_mentions = pull_mentions_direct
+
+    data_dicts = []
+    if run_title:
+        data_dicts.append(pull_mentions(paths, title_blocks, "title", words, replace_dict))
+    if run_abstract:
+        data_dicts.append(pull_mentions(paths, abstract_blocks, "abstract", words, replace_dict))
+    if run_body:
+        data_dicts.append(pull_mentions(paths, body_blocks, "body", words, replace_dict))
+
+    summed_dictionary = data_dicts[0]
```

Comment:
what happens if `data_dicts` has a length of greater than 1? Everything after the first element gets thrown out?